### PR TITLE
Add canonical transcript assignment info to docs

### PR DESCRIPTION
**Pull request by inodb**
Related to https://github.com/cBioPortal/cbioportal/issues/9797. Explain how canonical transcript assignments work thru Genome Nexus. Also add property `genomenexus.isoform_override_source`

TODO:

- [x]  Add property (make mskcc default, but set it to uniprot for genie during )
- [x] Add documentation for said property
- [x] Add it to SUMMARY.md
- [x] Needs associated frontend PR that leverages the new property (https://github.com/cBioPortal/cbioportal-frontend/pull/4400)

Deployment:

- Needs to be deployed together with https://github.com/cBioPortal/cbioportal-frontend/pull/4400
- ~Need to add property `genomenexus.isoform_override_source=uniprot` to genie portals~ (Done: https://github.com/knowledgesystems/knowledgesystems-k8s-deployment/commit/d52cbcf1a91f7b19849667413e6ee46dd737ed8f)

### CODE REVIEW

File: docs/mutation-data-transcript-annotation.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
...amino acids). For example: parts of the gene that code for proteins (exons) can be included or excluded. Each of these different proteins is called an [isoform](https://en.wikipedia.org/wiki/Protein_isoform). A single mutation can change the amino acids in each of these these isoforms differently. E.g. in one isoform it might change a P to a T, but in the other isoform that particular exon does not get included and it is therefore not changing the amino acid sequence at all. In cBioPortal for convenience sake we assign a single gene symbol + protein change to each mutation. In several cases this works well because there is only one protein isoform relevant in a clinical setting. There are of course exceptions and we are therefore working on improving this representation. The term isoform generally refers to a protein, but oftentimes one wants to know changes to non-exonic parts of the gene as well. Although cBioPortal currently only displays exonic events aside from a few exceptions of known oncogenic events (e.g. TERT promoter), we do store the change to the transcript in the MAF file. This allows computational researchers to easily investigate these further outside of cBioPortal.
```
Comment (Luke-Sikina) [inline docs/mutation-data-transcript-annotation.md:8]: > The term isoform generally refers to a protein, but oftentimes one wants to know changes to non-exonic parts of the gene as well. Although cBioPortal currently only displays exonic events aside from a few exceptions of known oncogenic events (e.g. TERT promoter), we do store the change to the transcript in the MAF file. This allows computational researchers to easily investigate these further outside of cBioPortal.

Do we need this information? It doesn't seem relevant to the following paragraphs. I was anticipating some sort of added complexity for non-exonic events, but you never mentioned it again.

**Review by Luke-Sikina — APPROVED** [review]

File: docs/mutation-data-transcript-annotation.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+## Mutation Data Transcript Annotation
```
[the remark this answers is not in the document]
Comment (inodb) [reply docs/mutation-data-transcript-annotation.md]: done

File: docs/mutation-data-transcript-annotation.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+[here](https://github.com/genome-nexus/genome-nexus-importer/blob/master/data/grch37_ensembl92/export/ensembl_biomart_canonical_transcripts_per_hgnc.txt).
+There are two sets of default transcripts: `uniprot` and `mskcc`. We recommend to use the `mskcc` set of transcripts when starting from scratch, since these are more up to date and correspond to transcripts that were chosen as relevant for clinical sequencing at MSKCC. The `uniprot` set of transcripts was constructed several years ago, but we are no longer certain about the logic on how to reconstruct them hence they are not being kept up to date. One can see the differences between the two in [this file](https://github.com/cBioPortal/cbioportal-frontend/files/9498680/genes_with_different_uniprot_mskcc_isoforms.txt). For the public cBioPortal (https://www.cbioportal.org) we are using `mskcc`, for for the GENIE cBioPortal (https://genie.cbioportal.org) we use `uniprot`. Although for local installations the default currently is `uniprot` to avoid people upgrading from a previous version of cBioPortal to accidentally switch to `mskcc`, we do recommend to use `mskcc`.
```
[the remark this answers is not in the document]
Comment (inodb) [reply docs/mutation-data-transcript-annotation.md]: agreed - need to still add this to properties etc

File: docs/mutation-data-transcript-annotation.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+#### What is a transcript?
+DNA is transcribed to an mRNA transcript which is then translated to a protein sequence. DNA sequencers sequence DNA which is why the transcript is usually stored as cDNA (the complementary DNA sequence of the mRNA) and the change is stored as a change to that cDNA transcript (the Ensembl database assigns ids for these transcript with names like `ENSTxxx`). For each gene name in cBioPortal a canonical transcript is assigned. These assignemnts are stored in [Genome Nexus](https://www.genomenexus.org/). Although cBioPortal does not store changes to different transcripts/isoforms for each mutation in the database itself, it does allow viewing them on the [Mutations Tab](https://bit.ly/39hVtDd) by re-annotating the mutations on the fly through [Genome Nexus](https://www.genomenexus.org/).
```
[the remark this answers is not in the document]
Comment (inodb) [reply docs/mutation-data-transcript-annotation.md]: i tried to illustrate this a bit better with some screenshots

File: docs/mutation-data-transcript-annotation.md (hunk 1 above)
Comment (inodb) [reply docs/mutation-data-transcript-annotation.md]: good point - i removed it

File: docs/mutation-data-transcript-annotation.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+### Transcript Assignment 
+
+The cBioPortal database stores one gene + protein change annotation for each mutation event in the database. To allow 
+comparing mutation data across studies it is important to annotate the mutation data (be it in MAF or VCF format) in 
+the same way, otherwise the gene + protein changes can mean entirely different things. For all public studies stored 
+in [datahub](https://github.com/cBioPortal/datahub/tree/master/public) we leverage [Genome Nexus](https:
+//www.genomenexus.org) to do so. Genome Nexus assigns one canonical Ensembl Transcript + gene name + protein change for 
+each mutation. You can find the mapping of hugo symbol to transcript id[here](https://github.com/genome-nexus/genome-
+nexus-importer/blob/master/data/grch37_ensembl92/export/ensembl_biomart_canonical_transcripts_per_hgnc.txt).There are
```
Comment (pieterlukasse) [inline docs/mutation-data-transcript-annotation.md:49]: remove line break? This link does not render correctly in markdown

File: docs/mutation-data-transcript-annotation.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+in [datahub](https://github.com/cBioPortal/datahub/tree/master/public) we leverage [Genome Nexus](https:
+//www.genomenexus.org) to do so. Genome Nexus assigns one canonical Ensembl Transcript + gene name + protein change for 
+each mutation. You can find the mapping of hugo symbol to transcript id[here](https://github.com/genome-nexus/genome-
+nexus-importer/blob/master/data/grch37_ensembl92/export/ensembl_biomart_canonical_transcripts_per_hgnc.txt).There are 
+two sets of default transcripts: `uniprot` and `mskcc`. We recommend to use the `mskcc` set of transcripts when 
+starting from scratch, since these are more up to date and correspond to transcripts that were chosen as relevant for 
+clinical sequencing at MSKCC. The `uniprot` set of transcripts was constructed several years ago, but we are no longer 
+certain about the logic on how to reconstruct them hence they are not being kept up to date. One can see the 
+differences between the two in [this file](https://github.com/cBioPortal/cbioportal-
+frontend/files/9498680/genes_with_different_uniprot_mskcc_isoforms.txt). For the public cBioPortal (https:
```
Comment (pieterlukasse) [inline docs/mutation-data-transcript-annotation.md:55]: broken link in markdown.

File: docs/mutation-data-transcript-annotation.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+//www.cbioportal.org) we are using `mskcc`, for for the GENIE cBioPortal (https://genie.cbioportal.org) we use `uniprot`
```
Comment (pieterlukasse) [inline docs/mutation-data-transcript-annotation.md:56]: why is GENIE using this? That sounds like a mistake in the context of previous paragraph.

File: docs/mutation-data-transcript-annotation.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+. Although for local installations the default currently is `uniprot` to avoid people upgrading from a previous version
```
Comment (pieterlukasse) [inline docs/mutation-data-transcript-annotation.md:57]: Maybe break this sentence down? Something like: 

> Although for local installations the default is still `uniprot` (to not break existing installations when upgrading), we do recommend `mskcc` for new installations. We do recommend existing installations to consider migrating to `mskcc` soon.

File: docs/mutation-data-transcript-annotation.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+because there is only one protein isoform relevant in a clinical setting. There are of course exceptions and we are 
+therefore working on improving this representation. An explanation of the relation between transcripts and protein 
+isoforms can be found in the next section. 
+
+#### What is a transcript? 
+
+DNA is transcribed to an mRNA transcript which is then translated to a protein sequence. A transcript is thus 
+associated with a specific protein isoform. DNA sequencers sequence DNA which is why the transcript is usually stored 
+as cDNA (the complementary DNA sequence of the mRNA) and the change is stored as a change to that cDNA transcript (the 
+Ensembl database assigns ids for these transcript with names like `ENSTxxx`). You can see this on e.g. the Ensembl
```
Comment (pieterlukasse) [inline docs/mutation-data-transcript-annotation.md:26]: maybe remove this part
> DNA sequencers sequence DNA which is why the transcript is usually stored 
as cDNA (the complementary DNA sequence of the mRNA) and the change is stored as a change to that cDNA transcript 

So just start the new sentence at "The Ensembl database assigns..."

File: docs/mutation-data-transcript-annotation.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+#### Plans for default transcripts 
+
+We are planning to move to a single set of default transcripts over time, previously `uniprot` was used for the public 
+facing portals and is currently still the default for new installations. Our plan is to use `mskcc` everywhere and 
+eventually we will most likely move to [MANE](https://www.ensembl.org/info/genome/genebuild/mane.html). MANE is only 
+available for grch38 and since most of our data is for grch37 this is currently not feasible. Whichever set of 
+transcripts you choose to use, make sure to indicate so in the [Genome Nexus Annotation Pipeline](https:
+//github.com/genome-nexus/genome-nexus-annotation-pipeline#maf-annotation) (`--isoform-override <mskcc or uniprot>`) 
+and put the same set of transcripts in the properties file of cBioPortal, such that the [Mutations Tab](https:
+//bit.ly/39hVtDd) will show the correct canonical transcript (currently defaults to `uniprot`). The re-annotation of
```
Comment (pieterlukasse) [inline docs/mutation-data-transcript-annotation.md:83]: space in link

File: docs/mutation-data-transcript-annotation.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+#### Plans for default transcripts 
+
+We are planning to move to a single set of default transcripts over time, previously `uniprot` was used for the public 
+facing portals and is currently still the default for new installations. Our plan is to use `mskcc` everywhere and 
+eventually we will most likely move to [MANE](https://www.ensembl.org/info/genome/genebuild/mane.html). MANE is only 
+available for grch38 and since most of our data is for grch37 this is currently not feasible. Whichever set of 
+transcripts you choose to use, make sure to indicate so in the [Genome Nexus Annotation Pipeline](https:
+//github.com/genome-nexus/genome-nexus-annotation-pipeline#maf-annotation) (`--isoform-override <mskcc or uniprot>`) 
+and put the same set of transcripts in the properties file of cBioPortal, such that the [Mutations Tab](https:
```
Comment (pieterlukasse) [inline docs/mutation-data-transcript-annotation.md:82]: a link to the relevant properties line or the doc page that mentions this property would be nice!

**Review by pieterlukasse — COMMENTED** [review]
thanks @inodb 

Great document overall! I like the clear explanation. Well captured 👍 

I added a few suggestions for simplifying the text a bit. I also noticed many links had spaces, which makes them fail to render as such in markdown.

File: docs/mutation-data-transcript-annotation.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+This document describes how each mutation in cBioPortal gets annotated with a specific gene symbol + protein change. 
+
+### Biological Background 
+
+This section explains the concepts of protein isoforms and transcripts. 
+
+#### What is an isoform? 
+
+From a single gene (string of nucleotides) multiple protein sequences can be formed (string of amino acids). For 
+example: parts of the gene that code for proteins (exons) can be included or excluded. Each of these different proteins
```
Comment (alisman) [inline docs/mutation-data-transcript-annotation.md:12]: excluded ... depending upon xyz?

Comment (alisman) [reply docs/mutation-data-transcript-annotation.md:12]: each of these different [resulting] proteins is called ...

File: docs/mutation-data-transcript-annotation.md [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+### Biological Background 
+
+This section explains the concepts of protein isoforms and transcripts. 
+
+#### What is an isoform? 
+
+From a single gene (string of nucleotides) multiple protein sequences can be formed (string of amino acids). For 
+example: parts of the gene that code for proteins (exons) can be included or excluded. Each of these different proteins 
+is called an [isoform](https://en.wikipedia.org/wiki/Protein_isoform). A single mutation can change the amino acids in
```
Comment (alisman) [inline docs/mutation-data-transcript-annotation.md:13]: can impact the isoforms differently

File: docs/mutation-data-transcript-annotation.md [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+each of these these isoforms differently. E.g. in one isoform it might change a P to a T, but in the other isoform that 
+particular exon does not get included and it is therefore not changing the amino acid sequence at all. In cBioPortal 
+for convenience sake we assign a single gene symbol + protein change to each mutation. For most cases this works well 
+because there is only one protein isoform relevant in a clinical setting. There are of course exceptions and we are 
+therefore working on improving this representation. An explanation of the relation between transcripts and protein 
+isoforms can be found in the next section. 
+
+#### What is a transcript? 
+
+DNA is transcribed to an mRNA transcript which is then translated to a protein sequence. A transcript is thus
```
Comment (alisman) [inline docs/mutation-data-transcript-annotation.md:23]: This is a little unclear to me.  I think what you are saying is that the variability that results in multiple isoforms occurs during transcription process. I.e. multiple transcripts are possible from the same gene and those map 1 to 1 with isoforms?

File: docs/mutation-data-transcript-annotation.md [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+for convenience sake we assign a single gene symbol + protein change to each mutation. For most cases this works well 
+because there is only one protein isoform relevant in a clinical setting. There are of course exceptions and we are 
+therefore working on improving this representation. An explanation of the relation between transcripts and protein 
+isoforms can be found in the next section. 
+
+#### What is a transcript? 
+
+DNA is transcribed to an mRNA transcript which is then translated to a protein sequence. A transcript is thus 
+associated with a specific protein isoform. DNA sequencers sequence DNA which is why the transcript is usually stored 
+as cDNA (the complementary DNA sequence of the mRNA) and the change is stored as a change to that cDNA transcript (the
```
Comment (alisman) [inline docs/mutation-data-transcript-annotation.md:25]: "change" meaning the change that results in isoform?

File: docs/mutation-data-transcript-annotation.md [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+starting from scratch, since these are more up to date and correspond to transcripts that were chosen as relevant for 
+clinical sequencing at MSKCC. The `uniprot` set of transcripts was constructed several years ago, but we are no longer 
+certain about the logic on how to reconstruct them hence they are not being kept up to date. One can see the 
+differences between the two in [this file](https://github.com/cBioPortal/cbioportal-
+frontend/files/9498680/genes_with_different_uniprot_mskcc_isoforms.txt). For the public cBioPortal (https:
+//www.cbioportal.org) we are using `mskcc`, for for the GENIE cBioPortal (https://genie.cbioportal.org) we use `uniprot`
+. Although for local installations the default currently is `uniprot` to avoid people upgrading from a previous version 
+of cBioPortal to accidentally switch to `mskcc`, we do recommend to use `mskcc`. 
+
+#### How default transcript assignment affects the Mutations Tab
```
Comment (leexgh) [inline docs/mutation-data-transcript-annotation.md:60]: Probably can also mention how non-canonical transcripts affect the mutation tab? For example we hide Annotation column on non-canonical transcripts

File: docs/mutation-data-transcript-annotation.md (hunk 14 above)
Comment (inodb) [reply docs/mutation-data-transcript-annotation.md:23]: exactly - im not sure tho about the 1 to 1 cause I could imagine theoretically it's possible multiple transcripts produce the same isoform

File: docs/mutation-data-transcript-annotation.md (hunk 15 above)
Comment (inodb) [reply docs/mutation-data-transcript-annotation.md:25]: gonna remove this bit as Pieter suggested

File: docs/mutation-data-transcript-annotation.md (hunk 12 above)
Comment (inodb) [reply docs/mutation-data-transcript-annotation.md:12]: > excluded ... depending upon xyz?

this process is known as alternative splicing. I left it out originally but added a few more lines about it

File: docs/mutation-data-transcript-annotation.md (hunk 7 above)
Comment (inodb) [reply docs/mutation-data-transcript-annotation.md:56]: just legacy reasons - we are trying to upgrade to MSKCC but will take a bit

File: docs/mutation-data-transcript-annotation.md (hunk 16 above)
Comment (inodb) [reply docs/mutation-data-transcript-annotation.md:51]: added a note further down

File: docs/mutation-data-transcript-annotation.md (hunk 8 above)
Comment (inodb) [reply docs/mutation-data-transcript-annotation.md]: i played around with that sentence

File: docs/mutation-data-transcript-annotation.md (hunk 9 above)
Comment (inodb) [reply docs/mutation-data-transcript-annotation.md]: yeah makes sense

File: docs/mutation-data-transcript-annotation.md (hunk 11 above)
Comment (inodb) [reply docs/mutation-data-transcript-annotation.md]: added

**Review by alisman — APPROVED** [review]

**Pull request merged**